### PR TITLE
Feature/construct block response

### PR DESCRIPTION
**Pull request by Daniel-K-Ivanov**
**Detailed description**:
- Added `RetrieveLatest` function to the block repo interface
- Refactored the domain Block struct to use consensus start and consensus end
- Implemented `FromRosettaBlock` and `ToRosettaBlock`
- Implemented `ToRosettaOperation` (not final, still have some todos)
- Implemented `FromRosettaTransaction` and `ToRosettaTransaction`
- Enhances the Block queries to the DB (**error handling work still to be done**)
- Implemented `/block` **(excluding the error handling logic)**

**Which issue(s) this PR fixes**:
Related to #33 #35 #36 

**Special notes for your reviewer**:

### CODE REVIEW

File: hedera-mirror-rosetta/app/domain/types/block.go [hunk 1]
```diff
@@ -6,20 +6,44 @@ import (
...
+		t.FromRosettaTransaction(rosettaT)
+		tArray[i] = t
+	}
+	b.Transactions = tArray
 }
 
 // ToRosettaBlock returns Rosetta type Block from the current domain type Block
 func (b *Block) ToRosettaBlock() *rTypes.Block {
-	return nil
+	tArray := make([]*rTypes.Transaction, len(b.Transactions))
```
Comment (Perseverance) [inline hedera-mirror-rosetta/app/domain/types/block.go:37]: I'd name this `transactions` all lowercase

File: hedera-mirror-rosetta/app/domain/types/block.go [hunk 2]
```diff
@@ -6,20 +6,44 @@ import (
...
+	for i, t := range b.Transactions {
+		tArray[i] = t.ToRosettaTransaction()
+	}
+	rBlock := &rTypes.Block{
```
Comment (Perseverance) [inline hedera-mirror-rosetta/app/domain/types/block.go:41]: I'd just return &rTypes.Block {
...
}

and not use the variable rBlock

File: hedera-mirror-rosetta/app/domain/types/operation.go [hunk 3]
```diff
@@ -19,5 +21,21 @@ func (t *Operation) FromRosettaOperation(rOperation *rTypes.Operation) *Operatio
 }
 
 // ToRosettaOperation returns Rosetta type Operation from the current domain type Operation
-func (t *Operation) ToRosettaOperation(*rTypes.Operation) {
+func (t *Operation) ToRosettaOperation() *rTypes.Operation {
+	rOperation := rTypes.Operation{
+		OperationIdentifier: &rTypes.OperationIdentifier{
+			Index: t.Index,
+		},
+		RelatedOperations: []*rTypes.OperationIdentifier{}, //TODO
```
Comment (Perseverance) [inline hedera-mirror-rosetta/app/domain/types/operation.go:29]: I am not sure what I need TODO to finish it, can you add a bit more info on the TODO

Comment (Perseverance) [reply hedera-mirror-rosetta/app/domain/types/operation.go:29]: Also I am not sure if we will ever need to add Related Operation

File: hedera-mirror-rosetta/app/domain/types/transaction.go [hunk 4]
```diff
@@ -7,14 +7,32 @@ import (
...
+	Operations []*Operation
 }
 
 // FromRosettaTransaction populates domain type Transaction from Rosetta type Transaction
-func (t *Transaction) FromRosettaTransaction(rTransaction *rTypes.Transaction) *Transaction {
-	return nil // TODO Implement
+func (t *Transaction) FromRosettaTransaction(rTransaction *rTypes.Transaction) {
+	t.ID = rTransaction.TransactionIdentifier.Hash
+
+	oArray := make([]*Operation, len(rTransaction.Operations))
```
Comment (Perseverance) [inline hedera-mirror-rosetta/app/domain/types/transaction.go:17]: same comment as `tArray`. I do not think it is idiomatic to use the type of the variable and can be bad for readability

File: hedera-mirror-rosetta/app/domain/types/operation.go (hunk 3 above)
Comment (Daniel-K-Ivanov) [reply hedera-mirror-rosetta/app/domain/types/operation.go:29]: I just saw that it is not required by the API. However, my impressions (from the conversations we had) were that we would implement it.

Comment (Perseverance) [reply hedera-mirror-rosetta/app/domain/types/operation.go:29]: I think this has to do with the contract calls, etc. that we probably wont be using

**Pull request merged**